staging/src/k8s.io/apiserver/pkg/util/peerproxy/peerproxy.go

```diff
@@ -35,33 +42,46 @@ type Interface interface {
 }
 
 // New creates a new instance to implement unknown version proxy
-func NewPeerProxyHandler(informerFactory kubeinformers.SharedInformerFactory,
-	svm storageversion.Manager,
-	proxyTransport http.RoundTripper,
+// This method is used for an alpha feature UnknownVersionInteroperabilityProxy
+// and is subject to future modifications.
+func NewPeerProxyHandler(identityLeaseLabelSelector string,
+	informerFactory kubeinformers.SharedInformerFactory,
 	serverId string,
 	reconciler reconcilers.PeerEndpointLeaseReconciler,
-	serializer runtime.NegotiatedSerializer) *peerProxyHandler {
+	ser runtime.NegotiatedSerializer,
+	loopbackClientConfig *rest.Config,
+	proxyClientConfig *transport.Config) (*peerProxyHandler, error) {
 	h := &peerProxyHandler{
-		name:                  "PeerProxyHandler",
-		storageversionManager: svm,
-		proxyTransport:        proxyTransport,
-		svMap:                 sync.Map{},
-		serverId:              serverId,
-		reconciler:            reconciler,
-		serializer:            serializer,
+		name:                          "PeerProxyHandler",
+		serverId:                      serverId,
+		reconciler:                    reconciler,
+		serializer:                    ser,
+		loopbackClientConfig:          loopbackClientConfig,
+		proxyClientConfig:             proxyClientConfig,
+		localDiscoveryResponseCache:   make(map[schema.GroupVersion][]string),
+		localDiscoveryCacheTicker:     time.NewTicker(30 * time.Minute),
+		peerAggDiscoveryResponseCache: make(map[string]*peerAggDiscoveryInfo),
 	}
-	svi := informerFactory.Internal().V1alpha1().StorageVersions()
-	h.storageversionInformer = svi.Informer()
 
-	svi.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	h.apiserverIdentityInformer = informerFactory.Coordination().V1().Leases().Informer()
```

Comment:
I think you can filter the informer events by a label selector, that should save us from a lot of unnecessary events since most leases are renewed quite frequently.

Reply:
oh does that mean we actually dont need to check for the correct lease type (should be apiserver identity lease) in `isValidIdentityLease()` once we switch to this?

Reply:
okay I am now using `informers.NewFilteredLeaseInformer` to filter events only for the identity leases. Also kept `isValidIdentityLease()` the same as additional validation that we are operating on identity leases only.